[Bug] transpile fails in build workflow

```
UUID is "gnome-mosaic@jardon.github.com"
rm -rf _build target
env PROJECTS="floating_exceptions" ./scripts/transpile.sh
+ pwd
+ ***
+ rm -rf _build
+ mkdir -p _build/floating_exceptions
+ glib-compile-schemas schemas
+ tsc --p src/floating_exceptions
src/floating_exceptions/tsconfig.json:6:9 - error TS5011: The common source directory of 'tsconfig.json' is './src'. The 'rootDir' setting must be explicitly set to this or another path to adjust your output's file layout.
  Visit https://aka.ms/ts6 for migration information.

6         "outDir": "../../target/floating_exceptions",
          ~~~~~~~~

src/floating_exceptions/tsconfig.json:8:9 - error TS5101: Option 'downlevelIteration' is deprecated and will stop functioning in TypeScript 7.0. Specify compilerOption '"ignoreDeprecations": "6.0"' to silence this error.

8         "downlevelIteration": true,
          ~~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: src/floating_exceptions/tsconfig.json:6

make: *** [Makefile:45: compile] Error 2
Error: Process completed with exit code 2.
```

https://github.com/jardon/gnome-mosaic/actions/runs/24187708189/job/70596515667

--- a/.github/workflows/build.yaml
+++ b/.github/workflows/build.yaml
@@ -16,5 +16,7 @@ jobs:
               run: |
                   sudo apt update
                   sudo apt install -y make
-                  npm install typescript@latest
+                  npm install typescript@5.7.3
+            - name: Add node_modules to PATH
+              run: echo "${{ github.workspace }}/node_modules/.bin:$PATH" >> $GITHUB_PATH
             - run: make compile

--- a/.github/workflows/build.yaml
+++ b/.github/workflows/build.yaml
@@ -8,6 +8,13 @@ jobs:
         runs-on: ubuntu-latest
         steps:
             - uses: actions/checkout@v4
-            - name: Install typescript
-              run: sudo apt install -y node-typescript make
+            - name: Set up Node.js
+              uses: actions/setup-node@v4
+              with:
+                  node-version: '20'
+            - name: Install dependencies
+              run: |
+                  sudo apt update
+                  sudo apt install -y make
+                  npm install typescript@latest
             - run: make compile

--- a/.github/workflows/release.yaml
+++ b/.github/workflows/release.yaml
@@ -19,7 +19,9 @@ jobs:
               run: |
                   sudo apt update
                   sudo apt install -y make
-                  npm install typescript@latest
+                  npm install typescript@5.7.3
+            - name: Add node_modules to PATH
+              run: echo "${{ github.workspace }}/node_modules/.bin:$PATH" >> $GITHUB_PATH
             - name: Run build and create ZIP file
               run: make zip-file
             - name: Get tag name

--- a/Makefile
+++ b/Makefile
@@ -93,7 +93,9 @@ update-repository:
 	git clean -fd
 
 zip-file: all
-	cd _build && zip -qr "../$(UUID)_$(VERSION).zip" .
+	cd _build
+	rm schemas/gschemas.compiled
+	zip -qr "../$(UUID)_$(VERSION).zip" .
 
 .NOTPARALLEL: debug local-install
 

--- a/src/floating_exceptions/tsconfig.json
+++ b/src/floating_exceptions/tsconfig.json
@@ -4,6 +4,7 @@
         "target": "es2015",
         // "strict": true,
         "outDir": "../../target/floating_exceptions",
+        "rootDir": "./src",
         "forceConsistentCasingInFileNames": true,
         "lib": ["es2015", "dom"],
         "pretty": true,

--- a/src/floating_exceptions/tsconfig.json
+++ b/src/floating_exceptions/tsconfig.json
@@ -5,7 +5,6 @@
         // "strict": true,
         "outDir": "../../target/floating_exceptions",
         "forceConsistentCasingInFileNames": true,
-        "downlevelIteration": true,
         "lib": ["es2015", "dom"],
         "pretty": true,
         "removeComments": true,

